records_mover/session.py

```diff
@@ -99,17 +106,22 @@ def _infer_creds(session_type: str) -> BaseCreds:
                          f"{session_type}.")
 
 
+# This is a mypy-friendly way of doing a singleton object:
+#
+# https://github.com/python/typing/issues/236
 class PleaseInfer(Enum):
-    # This is a mypy-friendly way of doing a singleton object:
-    #
-    # https://github.com/python/typing/issues/236
+    token = 1
+
+
+class NotYetFetched(Enum):
     token = 1
```

Comment:
"None" for a cached cred means "no cred is available".  NotYetFetched means "we haven't checked yet".